Paracha_Junaid/Assignments/Flask_SQL/login_Val/server.py

```diff
@@ -0,0 +1,152 @@
+
+from flask import Flask, render_template, session, request, redirect, flash
+from flask.ext.bcrypt import Bcrypt
+import re
+from mysqlconnection import MySQLConnector
+
+EMAIL_REGEX = re.compile(r'^[a-zA-Z0-9.+_-]+@[a-zA-Z0-9._-]+\.[a-zA-Z]+$')
+PW_REGEX = re.compile(r'^(?=.*[0-9])(?=.*[A-Z])\w{8,}$')
+app = Flask(__name__)
+app.secret_key = 'my_secret_key'
+mysql = MySQLConnector(app, 'friends')
+bcrypt = Bcrypt(app)
+@app.route('/')
+def index():
+    session['Register_class'] = 'hide'
+    session['Login_class'] = 'show'
+    if 'user' in session:
+        return redirect('/logged_in')
+    return render_template('index.html')
+
+@app.route('/login', methods = ['POST'])
+def login():
+    print request.form
+    if not 'action' in request.form:
+        flash('we got a problem...')
+        print "Ibrokere"
+        return redirect('/')
+    valid = True
+    if not EMAIL_REGEX.match(request.form['email']):
+        flash("User name is invalid!")
+        valid = False
+        print valid        
+        return redirect('/')
+    elif not PW_REGEX.match(request.form['password']):
+        flash("Password is invalid!")        
+        valid = False        
+        return redirect('/')
+    else: 
+        Str = "SELECT * FROM user WHERE email= :email"
+        data = {
+                 'email':request.form['email'], 
+                }
+        user_data = mysql.query_db(Str,data)
+        print user_data
+        if user_data != []:
+            print 'point1'
+            session['user info'] = user_data[0]
```

Comment:
A)  you should check against the user_data variable first before saving anything into session
B)  only save into session the info you need.  By saving the entire thing you've also saved the hashed password into session